store/postgres/user.go

```diff
@@ -18,37 +20,194 @@ import (
+			), goqu.C("id").Gt(uuid),
+		)).Order(goqu.C("id").Asc()).Limit(uint(limit)).ToSQL()
+	} else {
+		// Previous Page Query
+		listUsersInnerQuery := dialect.From("users").Where(goqu.And(
+			goqu.Or(
+				goqu.C("name").ILike(fmt.Sprintf("%%%s%%", keyword)),
+				goqu.C("email").ILike(fmt.Sprintf("%%%s%%", keyword)),
+			), goqu.C("id").Lt(uuid),
+		)).Order(goqu.C("id").Desc()).Limit(uint(limit))
+
+		listUsersQuery, _, err = goqu.From(listUsersInnerQuery).Order(goqu.C("id").Asc()).ToSQL()
+	}
+
+	return listUsersQuery, err
+}
+
+func buildSelectUserForUpdateQuery(dialect goqu.DialectWrapper) (string, error) {
+	selectUserForUpdateQuery, _, err := dialect.From("users").Prepared(true).Where(
+		goqu.Ex{
+			"id": "id_PH",
+		}).ToSQL()
+
+	return selectUserForUpdateQuery, err
+}
+
+func buildUpdateUserQuery(dialect goqu.DialectWrapper) (string, error) {
+	updateUserQuery, _, err := dialect.Update("users").Set(
+		goqu.Record{
+			"name":       goqu.L("$2"),
+			"email":      goqu.L("$3"),
+			"metadata":   goqu.L("$4"),
+			"updated_at": goqu.L("now()"),
+		}).Where(goqu.Ex{
+		"id": goqu.L("$1"),
+	}).Returning(&User{}).ToSQL()
+
+	return updateUserQuery, err
+}
+
+func buildUpdateCurrentUserQuery(dialect goqu.DialectWrapper) (string, error) {
+	updateCurrentUserQuery, _, err := dialect.Update("users").Set(
+		goqu.Record{
+			"name":       goqu.L("$2"),
+			"metadata":   goqu.L("$3"),
+			"updated_at": goqu.L("now()"),
+		}).Where(goqu.Ex{
+		"email": goqu.L("$1"),
+	}).Returning(&User{}).ToSQL()
+
+	return updateCurrentUserQuery, err
+}
+
+func buildListUserGroupsQuery(dialect goqu.DialectWrapper) (string, error) {
+	listUserGroupsQuery, _, err := dialect.Select(
+		goqu.I("g.id").As("id"),
+		goqu.I("g.metadata").As("metadata"),
+		goqu.I("g.name").As("name"),
+		goqu.I("g.slug").As("slug"),
+		goqu.I("g.updated_at").As("updated_at"),
+		goqu.I("g.created_at").As("created_at"),
+		goqu.I("g.org_id").As("org_id"),
+	).From(goqu.L("relations r")).
+		Join(goqu.L("groups g"), goqu.On(
+			goqu.I("g.id").Cast("VARCHAR").
+				Eq(goqu.I("r.object_id")),
+		)).Where(goqu.Ex{
+		"r.object_namespace_id": definition.TeamNamespace.Id,
+		"subject_namespace_id":  definition.UserNamespace.Id,
+		"subject_id":            goqu.L("$1"),
+		"role_id":               goqu.L("$2"),
+	}).ToSQL()
+
+	return listUserGroupsQuery, err
+}
+
+// Helper Function
+func (s Store) getLimits(ctx context.Context, label string, uuid string, keyword string) (string, error) {
+	var fetchedUser User
+
+	pageToken := PageTokenizer(label, uuid)
+	query, err := buildListUsersQuery(dialect, 1, pageToken, keyword)
+	if err != nil {
+		return "", fmt.Errorf("%w: %s", queryErr, err)
+	}
+
+	err = s.DB.WithTimeout(ctx, func(ctx context.Context) error {
+		return s.DB.GetContext(ctx, &fetchedUser, query)
+	})
+
+	var labelId string
+	if errors.Is(err, sql.ErrNoRows) {
+		if label == "next" {
+			labelId = "ffffffff-ffff-ffff-ffff-ffffffffffff"
```

Comment:
how about even this can be extracted out `ffffffff-ffff-ffff-ffff-ffffffffffff`... wdys ?